src/Storage/Helpers/DataElementHelper.cs

```diff
@@ -38,13 +40,30 @@ public static DataElement CreateDataElement(string dataType, List<Guid> refs, In
                 LastChanged = creationTime,
                 Size = fileSize,
                 Refs = refs,
-            };            
+            };
+
+            if (generatedFromId != null && generatedFromId.Any())
```

Comment:
Use the List.Count before IEnumeration.Any().

Reply:
I keep forgetting that Any is smart and can detect that the underlying type is a List. Your choice.

Reply:
Last comment. It's technically unnecessary to check for elements before a foreach. It's built in.

Reply:
but we avoid instantiating the references list if we don't have any references to include, so I guess the check still makes sense?

Reply:
Yeah, keep it.

With that being said there is another side to that discusion we should take one day. The nullable discussion. The rules say that a List reference should never be null. Just nitpicking. 😄